adr/BlockNoteBundle.ts

```diff
@@ -0,0 +1,80 @@
+import { BlockNoteBundle } from "./BlockNoteExtension";
+
+export interface BlockNoteBundleConfig<
+  Schema,
+  Extensions extends BlockNoteExtension<any>[],
+> {
+  /**
+   * The prosemirror plugins of the extension
+   */
+  extensions?: Extensions;
+
+  /**
+   * The schema that this extension adds to the editor
+   */
+  schema?: BlockNoteSchema<Schema>;
+
+  /**
+   * Add a dictionary of locales to the editor
+   */
+  dictionary?: Record<string, Locale>;
+
+  /**
+   * Items that are available to the slash menu
+   */
+  slashMenuItems?: (
+    | {
+        name: string;
+        icon?: any;
+        onClick: () => void;
+      }
+    // or return a component
+    | (() => any)
+  )[];
+
+  /**
+   * Items that are available to the formatting toolbar
+   */
+  formattingToolbar?: (
+    | {
+        name: string;
+        icon?: any;
+        onClick: () => void;
+      }
+    // or return a component
+    | (() => any)
+  )[];
+}
+
+export type Locale = Record<string, string | Record<string, string>>;
+
+/**
+ * This is an example of what an extension might look like
+ */
+export function multiColumnExtension(extensionOptions: {
```

Comment:
I noticed this is quite a different approach to how we currently do e.g. the AI extension, as here everything is v much bundled together whereas for AI you add stuff like slash menu items manually.

This seems like another case of convenience vs composability, as it's much nicer to add an extension but can be annoying if e.g. you don't want to include the extension's keyboard handlers. What made you lean towards this approach instead?